### PR TITLE
feat(j-s): Use new ids

### PR DESCRIPTION
**Pull request by unakb**
# Use new ids

[Ný innskráning island.is](https://app.asana.com/0/1199153462262248/1204723923830375/f)

## What

Use new ids login page 

## Why

So users can login via app as well as phone number

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205314680853560

### CODE REVIEW

File: apps/judicial-system/api/src/app/modules/auth/auth.controller.ts [hunk 1]
```diff
@@ -146,12 +103,89 @@ export class AuthController {
...
+            res,
+            redirectRoute,
+            new Entropy({ bits: 128 }).string(),
+          )
+        }
+        if (!verifiedToken) {
+          this.logger.debug('Token verification unsuccessful')
+        }
+      } catch (error) {
+        this.logger.debug('Token verification failed')
```
Comment (unakb) [inline apps/judicial-system/api/src/app/modules/auth/auth.controller.ts:185]: log error

File: apps/judicial-system/api/src/app/modules/auth/auth.controller.ts [hunk 2]
```diff
@@ -146,12 +103,89 @@ export class AuthController {
...
+            {
+              nationalId: token.nationalId,
+            },
+            res,
+            redirectRoute,
+            new Entropy({ bits: 128 }).string(),
+          )
+        }
+        if (!verifiedToken) {
+          this.logger.debug('Token verification unsuccessful')
```
Comment (unakb) [inline apps/judicial-system/api/src/app/modules/auth/auth.controller.ts:182]: log error

**Review by gudjong — APPROVED** [review]

File: apps/judicial-system/api/infra/judicial-system-api.ts [hunk 3]
```diff
@@ -7,15 +7,20 @@ export const serviceSetup = (services: {
...
+      AUTH_IDS_SCOPE: 'openid profile',
+      AUTH_IDS_CLIENT_ID: '@rettarvorslugatt.island.is/web',
+      AUTH_IDS_DOMAIN: {
+        dev: 'https://identity-server.dev01.devland.is',
+        staging: 'https://identity-server.staging01.devland.is',
+        prod: 'https://innskra.island.is',
       },
-      AUTH_AUDIENCE: {
-        dev: ref((h) => `judicial-system.${h.env.domain}`),
-        staging: ref((h) => `judicial-system.${h.env.domain}`),
```
Comment (AndesKrrrrrrrrrrr) [inline apps/judicial-system/api/infra/judicial-system-api.ts:17]: I'm not too fond of hard-coded URLs. This change will probably break feature-deployments.

File: apps/judicial-system/api/src/environments/environment.ts
Comment (AndesKrrrrrrrrrrr) [inline apps/judicial-system/api/src/environments/environment.ts:1]: What is this file used for? Why isn't `apps/judicial-system/api/infra/judicial-system-api.ts` enough?

Comment (unakb) [reply apps/judicial-system/api/src/environments/environment.ts:1]: This is just legacy config that we're phasing out

File: apps/judicial-system/api/infra/judicial-system-api.ts (hunk 3 above)
Comment (unakb) [reply apps/judicial-system/api/infra/judicial-system-api.ts:17]: Could you elaborate, do you mean rather doing something like this? :
 dev: ref(
          (h) =>
            `https://judicial-system.${h.env.domain}/api/auth/callback/identity-server`,
        ),
        
        Or do you mean they should be kept somewhere else like the AWS parameter store?

Comment (AndesKrrrrrrrrrrr) [reply apps/judicial-system/api/infra/judicial-system-api.ts:17]: Yes, that should work. Can we try it out and make a feature deployment to be sure?

File: apps/judicial-system/api/infra/judicial-system-api.ts [hunk 4]
```diff
@@ -7,15 +7,24 @@ export const serviceSetup = (services: {
...
-        prod: 'rettarvorslugatt.island.is',
+      AUTH_IDS_REDIRECT_URI: {
+        dev: ref(
+          (h) =>
+            `https://judicial-system.${h.env.domain}/api/auth/callback/identity-server`,
+        ),
+        staging: ref(
+          (h) =>
+            `https://judicial-system.${h.env.domain}/api/auth/callback/identity-server`,
+        ),
```
Comment (AndesKrrrrrrrrrrr) [inline apps/judicial-system/api/infra/judicial-system-api.ts:25]:
```suggestion
        dev: ref(
          (ctx) =>
            `${
              ctx.featureDeploymentName ? `${ctx.featureDeploymentName}-` : ''
            }judicial-system.dev01.devland.is`,
          ),
        staging: ref(
          (h) =>
            `https://identity-server.staging01.devland.is/api/auth/callback/identity-server`,
        ),
```

**Review by AndesKrrrrrrrrrrr — COMMENTED** [review]
Other than my suggested change to support feature-deployments, LGTM

File: apps/judicial-system/api/infra/judicial-system-api.ts (hunk 4 above)
Comment (unakb) [reply apps/judicial-system/api/infra/judicial-system-api.ts]: I've applied this change

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]

**Pull request merged**